src/Box.tsx

```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+export default function Box(props: JSX.IntrinsicElements["mesh"]) {
+    // Hold state for hovered and clicked events
+    const [hovered, setHovered] = useState(false);
+    const [clicked, setClicked] = useState(false);
+
+    return (
+        <mesh
+            {...props}
+            scale={clicked ? 1.5 : 1}
+            onClick={(event) => {
+                event.stopPropagation();
+                setClicked(!clicked);
+            }}
+            onPointerOver={(event) => {
+                event.stopPropagation();
+                setHovered(true);
+            }}
+            onPointerOut={(_) => setHovered(false)}
+        >
+            <boxGeometry args={[1, 1, 1]} />
+            <meshStandardMaterial color={hovered ? "hotpink" : "orange"} />
+        </mesh>
+    );
```

Comment:
nit: this file is using 4 spaces for indentation and the others are using 2. You may want to check your IDE settings.

Reply:
Ah ok thanks. I'll update this